app/controllers/application_controller.rb

```diff
@@ -112,34 +120,49 @@ def store_oauth_token(json = {})
     session[:endpoint] = json['endpoint']
   end
 
+  def store_session_data(json = {})
+    # Launchpad login
+    session[:launchpad_cookie] = json['launchpad_cookie']
+    session[:auid] = json['auid']
+    session[:launchpad_email] = json['launchpad_email']
+    session[:logged_in_at] = json.empty? ? nil : Time.now.to_i
+    session[:expires_in] = json.empty? ? 900 : json['expires_in'] # 15 min - Launchpad default time
+  end
+
   def logged_in?
-    if urs_login_required?
+    if launchpad_login_required?
+      is_user_logged_in = session[:launchpad_cookie].present? &&
+                          session[:auid].present? &&
+                          session[:urs_uid].present? &&
+                          session[:expires_in].present? &&
+                          session[:logged_in_at].present?
+
+      store_session_data unless is_user_logged_in # clear session token and info if user is not logged in
+    elsif urs_login_required?
       is_user_logged_in = session[:access_token].present? &&
                           session[:refresh_token].present? &&
                           session[:expires_in].present? &&
                           session[:logged_in_at].present?
 
       store_oauth_token unless is_user_logged_in # clear session time and token values if user is not logged in
-      is_user_logged_in
     end
 
+    is_user_logged_in
   end
 
   def is_logged_in
-    if !urs_login_required? # && !launchpad_login_required?
-      # TODO this should only happen if URS login AND Launchpad login are both turned off (which should not happen)
-      Rails.logger.error('An error has occured. Both URS and Launchpad feature toggle environment variables seem to have been turned off. Please check urs_login_required and launchpad_login_required')
+    ensure_one_login_method or return
 
-      redirect_to root_url, flash: { error: "An error has occurred with our login system. Please contact #{view_context.mail_to('support@earthdata.nasa.gov', 'Earthdata Support')}." }
-    end
+    session[:return_to] = request.fullpath
 
-    if urs_login_required?
+    if launchpad_login_required?
+      Rails.logger.info("Launchpad #{launchpad_cookie_name} Cookie: #{token}") if Rails.env.development?
```

Comment:
move the `if Rails.env.development?` outside this statement since you call it no matter what